client/client.go

```diff
@@ -2129,8 +2166,11 @@ func (c *Client) allocSync() {
 
 			// Send to server.
 			args := structs.AllocUpdateRequest{
-				Alloc:        toSync,
-				WriteRequest: structs.WriteRequest{Region: c.Region()},
+				Alloc: toSync,
+				WriteRequest: structs.WriteRequest{
+					Region:    c.Region(),
+					AuthToken: c.secretNodeID(),
```

Comment:
My only concern here is that it doesn't *explicitly* block on registration. It *implicitly* blocks on registration because the `Client.run` method blocks and therefore no updates can occur until that's unblocked.

However if for some reason we change that behavior so that runners restart prior to registration, this will implicitly start sending un-authed AllocUpdateRequests.

I _think_ we should explicitly block here to enforce `no RPCs until secret is registered` as an invariant.

We could even change `secretNodeID() string` to be `secretNodeID() (secret string, shutdown bool)`, have it wait on `c.shutdownCh`, and return `"", true` if a shutdown occurred.

I don't know... I don't feel particularly strongly about this, but we do have a history of `change X + change Y much later = bug Z from emergent behavior`

Reply:
Resolved by the new RPC method code